deployment/registry.py

```diff
@@ -196,33 +199,49 @@ def merge_registries(
     deprecated_contracts = deprecated_contracts or []
 
     # Read the registries, excluding deprecated contracts
-    reg1 = {
-        e.name: e for e in read_registry(registry_1_filepath) if e.name not in deprecated_contracts
-    }
-    reg2 = {
-        e.name: e for e in read_registry(registry_2_filepath) if e.name not in deprecated_contracts
-    }
+    reg1 = defaultdict(OrderedDict)
+    reg2 = defaultdict(OrderedDict)
+
+    for e in read_registry(registry_1_filepath):
+        if e.name in deprecated_contracts:
+            continue
+        reg1[str(e.chain_id)][e.name] = e
```

Comment:
Handling integers as JSON keys here (it's not valid json if so)?

Reply:
No longer dealing with json at this point. I guess I am just using the integer as a python dictionary key. Perhaps I don't need to cast to string.

Reply:
✔️ - fixed. Just use the integer as is as keys for the python dictionary.